src/status_im/contexts/communities/actions/share_community_channel/view.cljs

```diff
@@ -11,9 +11,16 @@
 (defn view
   []
   (fn []
-    (let [{:keys [url chat-id]}           (rf/sub [:get-screen-params])
+    (let [params                          (rf/sub [:get-screen-params])
+          ;; NOTE(seanstrom): We need to store these screen params for when the modal closes
+          ;; because the screen params will be cleared.
+          {:keys [url chat-id]}           @(rn/use-ref-atom params)
```

Comment:
Since the screen params are present in re-frame state, can't we just fetch it from there whenever we need it ? 

Why create an extra copy?  What's the constraint here ?

Reply:
The screen params are stored and then cleared from re-frame state when this view is animated to a closed state. So what will happen is that the `url` and `chat-id` will be set to `nil`, and this will cause a visual glitch with the QR code for a brief moment.

The above code is trying to prevent this visual glitch by keeping a local copy of the screen params. This way when the screen params are cleared while navigating, the QR code will still visualised with the intended `url` value.

Reply:
Does this apply only to this QR code (the visual glitch) or all QR codes?